refactor!: renaming `#[internal]` as `#[only_self]` + updated tests (#17958)
AztecProtocol/aztec-packages: docs/docs/developers/docs/aztec-nr/framework-description/functions/attributes.md

- The Aztec macro processor skips the [insertion of the initialization check](#initializer-functions-initializer) for this specific function
- The function can be called at any time, even if the contract hasn't been initialized yet

## #[only_self]

> Functions marked with #[only_self] attribute can only be called by the contract itself.

Reviewer: ```suggestion
External functions marked with #[only_self] attribute can only be called by the contract itself - if other contracts try to make the call it will fail.
```

This attribute is commonly used when an action starts in private but needs to be completed in public. The public
function must be marked with #[only_self] to restrict access to only the contract itself. A typical example is a private
token mint operation that needs to enqueue a call to a public function to update the publicly tracked total token
supply.

> 

Reviewer: ```suggestion

It is also useful in private functions when dealing with tasks of an unknown size but with a large upper bound (e.g. when needing to process an unknown amount of notes or nullifiers) as they allow splitting the work in multiple circuits, possibly resulting in performance improvements for low-load scenarios.

```

This macro inserts a check at the beginning of the function to ensure that the caller is the contract itself. This is done by adding the following assertion:

```rust

> assert(context.msg_sender() == context.this_address(), "Function can only be called by the same contract");

Reviewer: ```suggestion
assert(self.msg_sender() == self.address, "Function can only be called by the same contract");
```
